src/main/java/org/breedinginsight/services/validators/TraitValidatorService.java

```diff
@@ -207,6 +208,23 @@ public ValidationErrors checkTraitFieldsLength(List<Trait> traits, TraitValidato
         return errors;
     }
 
+    public ValidationErrors checkTraitFieldsFormat(List<Trait> traits, TraitValidatorErrorInterface traitValidatorErrors) {
+
+        ValidationErrors errors = new ValidationErrors();
+
+        for (int i = 0; i < traits.size(); i++) {
+
+            Trait trait = traits.get(i);
+            String name = trait.getObservationVariableName();
+
+            if (name != null && name.contains(".")){
+                ValidationError error = traitValidatorErrors.getPeriodObsVarNameMsg();
+                errors.addError(traitValidatorErrors.getRowNumber(i), error);
+            }
```

Comment:
This check is fine as is, but since this validation step is checking the entire trait format, it might be nice to use a regex here so that it can be easily updated in the future to handle additional characters or edge cases.

```suggestion
            Pattern pattern = Pattern.compile("\\.");
            Matcher matcher = pattern.matcher(name);
            boolean containsInvalidCharacter = matcher.find();
            
            if (name != null && containsInvalidCharacter){
                ValidationError error = traitValidatorErrors.getPeriodObsVarNameMsg();
                errors.addError(traitValidatorErrors.getRowNumber(i), error);
            }
```